engine/src/main/java/org/terasology/rendering/opengl/OpenGLUtils.java

```diff
@@ -167,6 +167,20 @@ public static void renderFullscreenQuad() {
         glPopMatrix();
     }
 
+    /**
+     * First sets a viewport and then renders a quad filling it.
+     *
+     * @param x              an integer representing the x coordinate (in pixels) of the origin of the viewport.
+     * @param y              an integer representing the y coordinate (in pixels) of the origin of the viewport.
+     * @param viewportWidth  an integer representing the width (in pixels) the viewport.
+     * @param viewportHeight an integer representing the height (in pixels) the viewport.
+     */
+    // TODO: perhaps remove this method and make sure the viewport is set explicitely.
+    public static void renderFullscreenQuad(int x, int y, int viewportWidth, int viewportHeight) {
```

Comment:
Hmmm... where did this come from? This method is not actually _necessarily_ rendering a fullscreen quad. It renders a full viewport quad, which might or might not be full screen. In fact, even if it rendered to the whole display, it might still not be full screen as the display might be windowed. Can you find a way to clean this up? You might have to rename this and the other renderFullscreenQuad to a combination of setViewportToSizeOfDisplay() and renderViewportFillingQuad() or just renderQuad().